src/webgpu/api/operation/rendering/blending.spec.ts

```diff
@@ -7,7 +8,269 @@ TODO:
+  col: GPUColorDict,
+  factor: GPUBlendFactor
+): GPUColorDict {
+  switch (factor) {
+    case 'zero':
+      return { r: 0, g: 0, b: 0, a: 0 };
+    case 'one':
+      return { r: 1, g: 1, b: 1, a: 1 };
+    case 'src-color':
+      return { ...src };
+    case 'one-minus-src-color':
+      return mapColor(src, v => 1 - v);
+    case 'src-alpha':
+      return mapColor(src, () => src.a);
+    case 'one-minus-src-alpha':
+      return mapColor(src, () => 1 - src.a);
+    case 'dst-color':
+      return { ...dst };
+    case 'one-minus-dst-color':
+      return mapColor(dst, v => 1 - v);
+    case 'dst-alpha':
+      return mapColor(dst, () => dst.a);
+    case 'one-minus-dst-alpha':
+      return mapColor(dst, () => 1 - dst.a);
+    case 'src-alpha-saturated': {
+      const f = Math.min(src.a, 1 - src.a);
+      return { r: f, g: f, b: f, a: 1 };
+    }
+    case 'blend-color':
+      return { ...col };
+    case 'one-minus-blend-color':
+      return mapColor(col, v => 1 - v);
+  }
+}
+
+function computeBlendOperation(src: GPUColorDict, dst: GPUColorDict, operation: GPUBlendOperation) {
+  switch (operation) {
+    case 'add':
+      return mapColor(src, (v, k) => v + dst[k]);
+    case 'max':
+      return mapColor(src, (v, k) => Math.max(v, dst[k]));
+    case 'min':
+      return mapColor(src, (v, k) => Math.min(v, dst[k]));
+    case 'reverse-subtract':
+      return mapColor(src, (v, k) => dst[k] - v);
+    case 'subtract':
+      return mapColor(src, (v, k) => v - dst[k]);
+  }
+}
+
+g.test('GPUBlendComponent')
+  .desc(
+    `Test all combinations of parameters for GPUBlendComponent.
+
+  Tests that parameters are correctly passed to the backend API and blend computations
+  are done correctly by blending a single pixel. The test uses rgba32float as the format
+  to avoid checking clamping behavior (tested in api,operation,rendering,blending:clamp,*).
+
+  Params:
+    - component= {color, alpha} - whether to test blending the color or the alpha component.
+    - srcFactor= {...all GPUBlendFactors}
+    - dstFactor= {...all GPUBlendFactors}
+    - operation= {...all GPUBlendOperations}
+  `)
+  .cases(
+    params() //
+      .combine(poptions('component', ['color', 'alpha'] as const))
+      .combine(poptions('srcFactor', kBlendFactors))
+      .combine(poptions('dstFactor', kBlendFactors))
+      .combine(poptions('operation', kBlendOperations))
+  )
+  .fn(t => {
+    const textureFormat: GPUTextureFormat = 'rgba32float';
+    const srcColor: GPUColorDict = { r: 0.11, g: 0.61, b: 0.81, a: 0.44 };
+    const dstColor: GPUColorDict = { r: 0.51, g: 0.22, b: 0.71, a: 0.33 };
+    const varColor: GPUColorDict = { r: 0.91, g: 0.82, b: 0.73, a: 0.64 };
+
+    const srcFactor = computeBlendFactor(srcColor, dstColor, varColor, t.params.srcFactor);
+    const dstFactor = computeBlendFactor(srcColor, dstColor, varColor, t.params.dstFactor);
+
+    const expectedColor =
+      t.params.operation === 'min' || t.params.operation === 'max'
```

Comment:
computeBlendOperation already switches on the GPUBlendOperation, so maybe just fold this logic into it?

I didn't realize min/max were different. Confusing. Didn't see mention of this in D3D12 docs either.